src/__tests__/ChoiceInputField.test.tsx

```diff
@@ -0,0 +1,213 @@
+
+      expect(captionNode.getAttribute('id')).toBe(`${fieldId}-caption`)
+      expect(inputNode.getAttribute('aria-describedby')).toBe(`${fieldId}-caption`)
+    })
+  })
+
+  describe('warnings', () => {
+    it('should warn users if they do not pass an input', async () => {
+      const consoleSpy = jest.spyOn(global.console, 'warn')
+      render(
+        <SSRProvider>
+          <ChoiceInputField>
+            <ChoiceInputField.Label>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+            <ChoiceInputField.Caption>{CHECKBOXINPUTFIELD_CAPTION_TEXT}</ChoiceInputField.Caption>
+          </ChoiceInputField>
+        </SSRProvider>
+      )
+
+      expect(consoleSpy).toHaveBeenCalled()
+    })
+
+    it('should warn users if they pass an id directly to the input', async () => {
+      const consoleSpy = jest.spyOn(global.console, 'warn')
+      render(
+        <SSRProvider>
+          <ChoiceInputField>
+            <ChoiceInputField.Label>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+            <Checkbox id="testId" />
+            <ChoiceInputField.Caption>{CHECKBOXINPUTFIELD_CAPTION_TEXT}</ChoiceInputField.Caption>
+          </ChoiceInputField>
+        </SSRProvider>
+      )
+
+      expect(consoleSpy).toHaveBeenCalled()
+    })
+
+    it('should warn users if they pass a `disabled` prop directly to the input', async () => {
+      const consoleSpy = jest.spyOn(global.console, 'warn')
+      render(
+        <SSRProvider>
+          <ChoiceInputField>
+            <ChoiceInputField.Label>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+            <Checkbox disabled />
+            <ChoiceInputField.Caption>{CHECKBOXINPUTFIELD_CAPTION_TEXT}</ChoiceInputField.Caption>
+          </ChoiceInputField>
+        </SSRProvider>
+      )
+
+      expect(consoleSpy).toHaveBeenCalled()
+    })
+
+    it('should warn users if they pass a `required` prop directly to the input', async () => {
+      const consoleSpy = jest.spyOn(global.console, 'warn')
+      render(
+        <SSRProvider>
+          <ChoiceInputField>
+            <ChoiceInputField.Label>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+            <Checkbox required />
+            <ChoiceInputField.Caption>{CHECKBOXINPUTFIELD_CAPTION_TEXT}</ChoiceInputField.Caption>
+          </ChoiceInputField>
+        </SSRProvider>
+      )
+
+      expect(consoleSpy).toHaveBeenCalled()
+    })
+  })
+
+  it('should have no axe violations', async () => {
+    const {container} = render(
+      <SSRProvider>
+        <ChoiceInputField>
+          <ChoiceInputField.Label>{CHECKBOXINPUTFIELD_LABEL_TEXT}</ChoiceInputField.Label>
+          <Checkbox />
+          <ChoiceInputField.Caption>{CHECKBOXINPUTFIELD_CAPTION_TEXT}</ChoiceInputField.Caption>
+        </ChoiceInputField>
+      </SSRProvider>
+    )
+    const results = await axe(container)
+    expect(results).toHaveNoViolations()
+    cleanup()
```

Comment:
Might be worth running `cleanup` and `jest.clearAllMocks();` on an `afterEach` block at the top of the test suite, as you're using spy a fair bit here